src/pages/projects/ProjectsTable/ProjectsTable.test.tsx

```diff
@@ -0,0 +1,153 @@
+    const placeholder = 'Нет данных';
+    const table = renderComponent({ onFavorite: func, placeholder });
+
+    expect(table.getByText(placeholder)).toBeInTheDocument();
+  });
+
+  test('рендерится компонент placeholder', () => {
+    const func = jest.fn();
+    const placeholder = <div data-testid="placeholder">Нет данных</div>;
+    const table = renderComponent({ onFavorite: func, placeholder });
+
+    expect(table.getByTestId('placeholder')).toBeInTheDocument();
+  });
+
+  test('рендерится строка с данными', () => {
+    const func = jest.fn();
+    const table = renderComponent({ onFavorite: func, rows: projectRowMock });
+
+    expect(table.getByText(projectRowMock[0].name)).toBeInTheDocument();
+  });
+
+  test('добавляется аттрибут title для названия строки', () => {
+    const func = jest.fn();
+    const name = 'Lorem ipsum dolor sit amet consectetur adipisicing elit.';
+    const table = renderComponent({
+      onFavorite: func,
+      rows: [{ ...projectRowMock[0], name }],
+    });
+
+    expect(table.getByTitle(name)).toBeInTheDocument();
+  });
+
+  test('рендерится описание проекта', () => {
+    const func = jest.fn();
+    const description = 'Lorem ipsum dolor sit amet consectetur adipisicing elit.';
+    const table = renderComponent({
+      onFavorite: func,
+      rows: [{ ...projectRowMock[0], description }],
+    });
+
+    expect(table.getByText(description)).toBeInTheDocument();
+  });
+
+  test('отображается кнопка «избранное»', () => {
+    const func = jest.fn();
+    const table = renderComponent({ onFavorite: func, rows: projectRowMock });
+
+    tl.fireEvent.mouseOver(table.getByText(projectRowMock[0].name));
+
+    expect(table.getByTestId(ProjectsTable.testId.favoriteNotSelectedButton)).toBeInTheDocument();
+  });
+
+  test('отображается кнопка «избранное» в активном состоянии', () => {
+    const func = jest.fn();
+    const table = renderComponent({
+      onFavorite: func,
+      rows: [{ ...projectRowMock[0], isFavorite: true }],
+    });
+
+    tl.fireEvent.mouseOver(table.getByText(projectRowMock[0].name));
+
+    expect(table.getByTestId(ProjectsTable.testId.favoriteSelectedButton)).toBeInTheDocument();
+  });
+
+  test('вызывается onFavorite', () => {
+    const func = jest.fn();
+    const table = renderComponent({
+      onFavorite: func,
+      rows: projectRowMock,
+    });
+
+    tl.fireEvent.mouseOver(table.getByText(projectRowMock[0].name));
+    tl.fireEvent.click(table.getByTestId(ProjectsTable.testId.favoriteNotSelectedButton));
+
+    expect(func).toBeCalled();
+  });
+
+  test('отображается кнопка «меню»', () => {
+    const func = jest.fn();
+    const table = renderComponent({ onFavorite: func, rows: projectRowMock });
+
+    tl.fireEvent.mouseOver(table.getByText(projectRowMock[0].name));
+
+    expect(table.getByTestId(ProjectsTable.testId.favoriteNotSelectedButton)).toBeInTheDocument();
+  });
+
+  test('при наличии меню, строка становится активной', () => {
```

Comment:
```suggestion
  test('при открытом меню, строка становится активной', () => {
```

или "при активном меню"

Reply:
исправил